functions/src/main/java/com/example/functions/HelloGcsSample.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START functions_helloworld_storage]
+package com.example.functions;
+
+import java.util.logging.Logger;
+
+public class HelloGcsSample {
+  private static Logger logger = Logger.getLogger(HelloGcsSample.class.getName());
+
+  public void helloGcs(GcsEvent event) throws Exception {
```

Comment:
This doesn't match the servlet signature.

Reply:
My concern here is that Java has a very strict typing on classes, so I'm concerned that GCF isn't actually going to use a `com.example.functions.GcsEvent` class and this sample won't work.

Reply:
We plan to have the Java GSON parser to unmarshall the GcsEvent Pojo defined along with the user function. This experience aligns with the experience we currently provide users for Go. We also plan to open source the Java functions framework at Beta, so at that point an open source event type could be considered. However, I don't think this would be something we would consider as a blocker for Beta and GA.

Reply:
Is the class going to be derived from the parameter for the function then? 

Presumably if so, it would be safe to move `GcsEvent` as a static class inside the class with the function. This would allow us to keep the goal of having the sample copy/pastable.